libexpory-1.5 fails to build with uclibc

when compiling libpoxy-1.5 with uclibc, the build breaks with this error: 

```../libepoxy-1.5.2/test/egl_has_extension_nocontext.c: In function 'main':
../libepoxy-1.5.2/test/egl_has_extension_nocontext.c:58:24: error: implicit declaration of function 'strdup' [-Werror=implicit-function-declaration]
         an_extension = strdup(extensions);
                        ^~~~~~
../libepoxy-1.5.2/test/egl_has_extension_nocontext.c:58:24: warning: incompatible implicit declaration of built-in function 'strdup'
```
I added the build log and a log with the most important os libs and compilers. 

[full build log](https://github.com/anholt/libepoxy/files/2109973/build.log.zip)
[emerge --info](https://github.com/anholt/libepoxy/files/2109992/emerge-info.zip)

libepoxy 1.4 is just fine by the way

--- a/test/egl_epoxy_api.c
+++ b/test/egl_epoxy_api.c
@@ -27,6 +27,7 @@
  * Tests the Epoxy API using EGL.
  */
 
+#define _GNU_SOURCE
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>

--- a/test/egl_has_extension_nocontext.c
+++ b/test/egl_has_extension_nocontext.c
@@ -28,6 +28,7 @@
  * no context bound would fail out in dispatch.
  */
 
+#define _GNU_SOURCE
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>

